src/components/Header.tsx

```diff
@@ -0,0 +1,88 @@
+import classNames from 'classnames';
+import { forwardRef, MutableRefObject, useEffect, useState } from 'react';
+import { Todo } from '../types';
+
+type Props = {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  onAdd: (title: string) => void;
+  onError: (message: string) => void;
+  isError: boolean;
+  isLoading: boolean;
+  isAllCompleted: boolean;
+  onToggleAll: () => void;
+};
+
+// eslint-disable-next-line react/display-name
+export const Header = forwardRef<HTMLInputElement, Props>(
+  (
+    {
+      todos,
+      tempTodo,
+      onAdd,
+      onError,
+      isError,
+      isLoading,
+      isAllCompleted,
+      onToggleAll,
+    },
+    inputRef,
+  ) => {
+    const [title, setTitle] = useState('');
+
+    useEffect(() => {
+      if (inputRef && tempTodo === null) {
+        (inputRef as MutableRefObject<HTMLInputElement>).current.focus();
+      }
+    }, [tempTodo, inputRef]);
+
+    const handleSubmit = async (e: React.FormEvent) => {
+      e.preventDefault();
+
+      const trimmedTitle = title.trim();
+
+      if (trimmedTitle === '') {
+        onError('Title should not be empty');
+        setTimeout(() => onError(''), 3000);
+
+        return;
+      }
+
+      onAdd(trimmedTitle);
+    };
+
+    useEffect(() => {
+      if (!isLoading && !isError) {
+        setTitle('');
+      }
+    }, [isLoading, isError]);
+
+    return (
+      <header className="todoapp__header">
+        {!isLoading && todos.length > 0 && (
```

Comment:
```suggestion
        {!isLoading && todos.length && (
```

Reply:
Oh no. No way. I don't want the '0' inscription in my layout. I specifically fixed it earlier. Check the commit history:
5716d63ddadea21237f2b7a6d1fff2286540ba7b

Reply:
Okay